website/content/docs/connect/proxies/envoy-extensions/usage/lua.mdx

```diff
@@ -0,0 +1,185 @@
+---
+layout: docs
+page_title: Lua Envoy Extension
+description: >-
+  Learn how the `lua` Envoy extension enables Consul to run Lua scripts during Envoy requests and responses from Consul-generated Envoy resources.
+---
+
+# Run Lua scripts in Envoy proxy
+
+The Lua Envoy extension enables the HTTP Lua filter in your Consul Envoy proxies, letting you run Lua scripts when requests and responses pass through Consul-generated Envoy resources.
+
+Envoy filters support setting and getting dynamic metadata, allowing a filter to share state information with subsequent filters. To set dynamic metadata, configure the HTTP Lua Filter. Users can call `streamInfo:dynamicMetadata()` from Lua scripts to get the request's dynamic metadata. 
```

Comment:
This seems like a good place to link to <https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter> so that readers know what functions are available to them.

Reply:
```suggestion
Envoy filters support setting and getting dynamic metadata, allowing a filter to share state information with subsequent filters. To set dynamic metadata, configure the [HTTP Lua Filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter). Users can call `streamInfo:dynamicMetadata()` from Lua scripts to get the request's dynamic metadata. 
```